add trusted and remake information to RSS feed

nallown opened the pull request.

Manjimon commented:
This would be nice, yeah. Accept it thanks~

disturbed01 commented:
this looks pretty sweet, will simplify my life. please accept this

TheAMM commented on nyaa/templates/rss.xml:
```diff
@@ -40,6 +40,11 @@
+			<nyaa:submitter> {{- 'Anonymous' if torrent.anonymous or not torrent.user else torrent.user.username }}</nyaa:submitter>
+			<nyaa:trusted>	 {{- torrent.trusted }}</nyaa:trusted>
+			<nyaa:remake>	 {{- torrent.remake }}</nyaa:remake>
+			<nyaa:summary><![CDATA[{{- torrent.description or '#### No description.' }}]]></nyaa:summary>
```
Not available on [ES results](https://github.com/nyaadevs/nyaa/blob/master/es_mapping.yml#L84).

nallown replied:
How can I make available on Elastic Search?

TheAMM replied:
You can't. We don't store the description in ES, and it's not really worth the overhead to add it just for RSS.

nallown replied:
Should I make descriptions only available in none ES queries?

TheAMM replied:
Nah. It breaks consistency, pure-db queries are rate for RSS, few would have any use for the field (it's not visible for human-RSS readers), plus it's unprocessed markdown.

nallown replied:
You're right people don't really need the summary since they can use the other info from the feed. I'll remove it then since I don't need it either.

What should I do about your other change request? The one for the submitter field?

TheAMM commented on nyaa/templates/rss.xml:
```diff
@@ -40,6 +40,11 @@
 			<nyaa:categoryId>{{- cat_id }}</nyaa:categoryId>
 			<nyaa:category>  {{- category_name(cat_id) }}</nyaa:category>
 			<nyaa:size>      {{- torrent.filesize | filesizeformat(True) }}</nyaa:size>
+			<nyaa:submitter> {{- 'Anonymous' if torrent.anonymous or not torrent.user else torrent.user.username }}</nyaa:submitter>
```
`torrent.user` won't work on ES - you'll have to work around it as above with the category id.

nallown replied:
I checked the code, on Elastic Search you use `torrent.sub_category_id` instead of `torrent.sub_category.id_as_string`. How did you make the `sub_category.id_as_string` value available in `torrent.sub_category_id`?

I'm guessing I need to do something similar with the torrents username. Something like:
```j2
{% set submitter = use_elastic and torrent.submitter or torrent.user.username %}
```

I'm not sure how I can map the username to `torrent.submitter` though

TheAMM replied:
We don't make anything look like anything. You can see [just above](https://github.com/nyaadevs/nyaa/pull/481/files/29ccd939f8ac0f906804d57ea41e3607c5795397#diff-0c21456d97f5b069933b89256c6c1573R39) how we choose either the ES category field or the DB model's relationship field. You don't have to map that back to anything, just use the variable you just `set`.

nallown replied:
I get that but like how's the fields made? Or where can I find the ES field for username?

nallown replied:
I checked the ES Mappings and I saw you have a `uploader_id` in there. Is that the user's username or his account ID?

TheAMM replied:
`torrent` is either `models.Torrent` or an ES object with the fields listed in https://github.com/nyaadevs/nyaa/blob/master/es_mapping.yml#L84 when `use_elastic` is `True`.

nallown replied:
I already looked at the ES mappings but found no field for username or submitter, what value am I meant to show on ES?

TheAMM replied:
Oh shit, you're right, I remembered falsely (since `uploader_id` is there). This will complicate things, since we'll have to decide whether or not to store the uploader name in ES. I'll get back on this later.

nallown replied:
Any updates? If its too much work then I can remove the "submitter" from the pull requests since its not really needed. If someone needs it then it can be added later in another pull requests.

nallown replied:
@TheAMM Can you give me update

TheAMM commented on nyaa/templates/rss.xml:
```diff
@@ -40,6 +40,11 @@
 			<nyaa:size>      {{- torrent.filesize | filesizeformat(True) }}</nyaa:size>
+			<nyaa:submitter> {{- 'Anonymous' if torrent.anonymous or not torrent.user else torrent.user.username }}</nyaa:submitter>
+			<nyaa:trusted>	 {{- torrent.trusted }}</nyaa:trusted>
+			<nyaa:remake>	 {{- torrent.remake }}</nyaa:remake>
```
These should rather be explicit strings, and instead of `True` and `False` I'd prefer `Yes` and `No` (`torrent.remake and 'Yes' or 'No'`).

nallown replied:
Fixed.

nallown commented:
@TheAMM I've changed the pull request and removed the "submitted" and "description". I've also made every other change that you asked me to make with the rest.
